### PR TITLE
NMS-15666: improve Minion kafka config instructions

### PR DESCRIPTION
**Pull request by dschlenk**
### All Contributors

* [X] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [X] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://opennms.atlassian.net/) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the Jira link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the Jira issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-15666

### CODE REVIEW

File: docs/modules/deployment/pages/minion/message-broker/kafka.adoc [hunk 1]
```diff
@@ -52,19 +52,38 @@ config:property-set bootstrap.servers my-kafka-ip-1:9092,my-kafka-ip-2:9092<1>
...
-<1> Connect to the specified Kafka nodes and adjust the IPs or FQDNs with the Kafka port (9092) accordingly.
+<1> Adjust the IPs or FQDNs and the Kafka port (9092) to match your broker configuration.
 
-If you set more than one Kafka node as `bootstrap.servers`. the driver attempts to connect to the first entry.
+If you set more than one Kafka node as `bootstrap.servers` the driver attempts to connect to the first entry.
 If that is successful, the whole broker topology will be discovered and will be known by the client.
 The other entries are used only if the connection to the first entry fails.
 
-IMPORTANT: Ensure you use the FQDN or IP for your Kafka nodes as configured as advertised listener.
+Any valid Kafka configuration property can be used.
```
Comment (Bonrob2) [inline docs/modules/deployment/pages/minion/message-broker/kafka.adoc:61]:
```suggestion
You can use any valid Kafka configuration property.
```

File: docs/modules/deployment/pages/minion/message-broker/kafka.adoc [hunk 2]
```diff
@@ -52,19 +52,38 @@ config:property-set bootstrap.servers my-kafka-ip-1:9092,my-kafka-ip-2:9092<1>
...
+      An example using SASL/SCRAM with TLS:
```
Comment (Bonrob2) [inline docs/modules/deployment/pages/minion/message-broker/kafka.adoc:62]:
```suggestion
The following example uses SASL/SCRAM with TLS:
```

File: docs/modules/deployment/pages/minion/message-broker/kafka.adoc [hunk 3]
```diff
@@ -52,19 +52,38 @@ config:property-set bootstrap.servers my-kafka-ip-1:9092,my-kafka-ip-2:9092<1>
...
+config:edit org.opennms.core.ipc.kafka
+config:property-set bootstrap.servers my-kafka-ip-1:9096,my-kafka-ip-2:9096
+config:property-set security.protocol=SASL_SSL
+config:property-set sasl.mechanism SCRAM-SHA-512
+config:property-set sasl.jaas.config 'org.apache.kafka.common.security.scram.ScramLoginModule required username="opennms-ipc" password="kafka";'
+config:update
+----
+
+NOTE: You can configure module-specific values for modules `sink`, `rpc`, and `twin` by replacing PID `org.opennms.core.ipc.kafka` in the `config:edit` command with `org.opennms.core.ipc.sink.kafka`, `org.opennms.core.ipc.rpc.kafka`, and `org.opennms.core.ipc.twin.kafka`, respectively.
+      Module-specific config takes precedence over common config specified in config for PID `org.opennms.core.ipc.kafka`.
```
Comment (Bonrob2) [inline docs/modules/deployment/pages/minion/message-broker/kafka.adoc:75]:
```suggestion
Module-specific configuration takes precedence over common configuration specified in config for PID `org.opennms.core.ipc.kafka`.
```
I'm not clear what "specified in config for PID" means. Should it be in "config:edit" command?

File: docs/modules/deployment/pages/minion/message-broker/kafka.adoc [hunk 4]
```diff
@@ -52,19 +52,38 @@ config:property-set bootstrap.servers my-kafka-ip-1:9092,my-kafka-ip-2:9092<1>
...
 
 Exit the Karaf shell with kbd:[Ctrl+d]
 
+By default, the minion will use a topic prefix of `OpenNMS`. If you chose to use a different prefix in core, for instance, to use a single Apache Kafka cluster with multiple {page-component-title} instances, change the topic prefix by setting the `org.opennms.instance.id` system property in `$MINION_HOME/etc/custom.system.properties`, like so:
```
Comment (Bonrob2) [inline docs/modules/deployment/pages/minion/message-broker/kafka.adoc:79]:
```suggestion
By default, the Minion will use a topic prefix of `OpenNMS`.
If you chose to use a different prefix in core (for instance, to use a single Apache Kafka cluster with multiple {page-component-title} instances), change the topic prefix by setting the `org.opennms.instance.id` system property in `$MINION_HOME/etc/custom.system.properties`:
```

File: docs/modules/deployment/pages/minion/message-broker/kafka.adoc [hunk 5]
```diff
@@ -52,19 +52,38 @@ config:property-set bootstrap.servers my-kafka-ip-1:9092,my-kafka-ip-2:9092<1>
...
+[source, properties]
+----
+org.opennms.instance.id=MyNMS<1>
+----
+
+<1> Change `MyNMS` to the value used by the core instance of {page-component-title} that this minion should be connected to.
```
Comment (Bonrob2) [inline docs/modules/deployment/pages/minion/message-broker/kafka.adoc:85]:
```suggestion
<1> Change `MyNMS` to the the core instance of {page-component-title} value that this Minion should connect to.
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
Some minor changes. Thank you for writing this up.

File: docs/modules/deployment/pages/minion/message-broker/kafka.adoc (hunk 3 above)
Comment (dschlenk) [reply docs/modules/deployment/pages/minion/message-broker/kafka.adoc:75]: Yeah, I wasn't sure how to phrase this. PID stands for Persistent Identity; it's a term from the OSGi bundle configuration standard, meaning the config file for a specific OSGi bundle. It's a useful shortcut for the handful of users like me who know what it is, but probably not great for a general audience. I'll rework it to remove that reference.

File: docs/modules/deployment/pages/minion/message-broker/kafka.adoc (hunk 5 above)
Comment (dschlenk) [reply docs/modules/deployment/pages/minion/message-broker/kafka.adoc:86]: you've got 'the' twice - I'll rework this in this spirit though.

**Review by Bonrob2 — APPROVED** [review]
LGTM! Thank you.

File: docs/modules/deployment/pages/minion/message-broker/kafka.adoc (hunk 5 above)
Comment (Bonrob2) [reply docs/modules/deployment/pages/minion/message-broker/kafka.adoc]: Thanks. ARGH! I need to proofread my proofreading. :-)

**Review by mmahacek — APPROVED** [review]

**Pull request merged**